docs/DeveloperGuide.md

```diff
@@ -1,38 +1,682 @@
+or `Expense` type. Below is a simplified class diagram (with methods omitted) containing the attributes
+within each transaction and how each transaction is associated with the `TransactionList`.
+
+<p align="center">
+    <img src="images/TransactionClassDiagram.png">
+    <br />
+    <i>Figure 1: Simplified Class Diagram for Transaction</i>
+</p>
+
+The `TransactionList` holds a dynamic array list that can store multiple `Transaction` objects.
+
+Each Transaction object contains the following mandatory member attributes:
+
+1. **category** A category for the transaction.
+2. **description** More information regarding the transaction, written without any space.
+3. **amount** Value of the transaction in numerical form. Only integers within 0 and 10000000 is accepted.
+4. **date** Date of the transaction. The format must be in "yyyyMMdd".
+
+Some important operations are performed within the `TransactionList` class, which implements the following:
+
+- `TransactionList#addIncome(String description, int amount, String category, LocalDate date)` - Adds a transaction of
+  class type Income into the transactions list.
+- `TransactionList#addExpense(String description, int amount, String category, LocalDate date)` - Adds a transaction of
+  class type Expense into the transactions list.
+- `TransactionList#listTransactions(String type, String category, LocalDate date)` - List all or some transactions based
+  on selection.
+- `TransactionList#findTransactions(String keywords)` - Find specific transaction(s) based on any keywords inputted by
+  the user.
+- `TransactionList#deleteTransaction(int index)` - Deletes a transaction from the transactions list based on the
+  specified index.
+- `TransactionList#purgeTransactions()` - Purges all transactions in the transactions list.
+
+_Written by: Chua Han Yong Darren_
+
+### Help Command
+The help command displays the help message to the users to guide them on the usage and provide descriptions for each
+available command.
+
+The help command can be run as `help` or `help o/detailed`, where the latter will display a more detailed version of
+help messages to the users.
 
-## Design & implementation
+The structure of the application focusing on the help command is illustrated in the class diagram below:
+![Data Component Class Diagram](images/HelpClassDiagram.png)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+For each command subclass, they will implement the getHelpMessage() and getDetailedHelpMessage() methods. These methods
+will contain their corresponding HelpMessage Enum that stores the help messages as strings inside the enum.
 
+In the help command, during the execute() call, it will call either generateBasicHelp() or generateDetailedHelp() method
+based on the help option chosen by the user.
+
+![Data Component Class Diagram](images/HelpSequenceDiagram.png)
```

Comment:
The entry point for the application is presumably through Duke, so I think the User should be on the left hand side of the diagram providing the input and receiving the output by 'showHelp(message)'